crates/telio-relay/src/derp/http.rs

```diff
@@ -185,24 +190,33 @@ async fn connect_and_start<RW: AsyncRead + AsyncWrite + Send + 'static>(
 
     read_server_info(&mut reader).await?;
 
-    // Splitted channel for communication side and for connection side
-    let (comm_side, conn_side) = Chan::pipe();
+    // Split channels for communication side and for connection side. One channel for
+    // connections relayed through derp and another for communication with derp directly
+    let (comm_side_relayed, conn_side_relayed) = Chan::pipe();
+    let (comm_side_direct, conn_side_direct) = Chan::pipe();
 
-    let tx = conn_side.tx;
-    let rx = conn_side.rx;
+    let sender_relayed = conn_side_relayed.tx;
+    let receiver_relayed = conn_side_relayed.rx;
+    let sender_direct = conn_side_direct.tx;
+    let receiver_direct = conn_side_direct.rx;
 
     Ok(DerpConnection {
-        comms: comm_side,
+        comms_relayed: comm_side_relayed,
+        comms_direct: comm_side_direct,
         join_sender: tokio::spawn(async move {
-            start_read(reader, tx, addr)
+            start_read(reader, sender_relayed, sender_direct, addr)
                 .await
                 .map_err(|err| IoError::new(ErrorKind::Other, err.to_string()))
         }),
         join_receiver: tokio::spawn(async move {
-            start_write(writer, rx, addr)
+            start_write(writer, receiver_relayed, receiver_direct, addr)
                 .await
                 .map_err(|err| IoError::new(ErrorKind::Other, err.to_string()))
         }),
+        poll_timer: {
+            let poll_interval = Duration::from_secs(server_keepalives.derp_keepalive as u64);
+            interval_at(tokio::time::Instant::now() + poll_interval, poll_interval)
```

Comment:
Set missed tick behavior to [Delay](https://docs.rs/tokio/1.32.0/tokio/time/enum.MissedTickBehavior.html#variant.Delay)

Default behavior is Burst, meaning that after sleep you would fire sleep / poll_interval event's. This would be an unnecessary DDOS to relay servers :D